tutorials/arvo/arvo.md

```diff
@@ -0,0 +1,519 @@
+
+```
+~[
+  /g/a/~zod/4_shell_terminal/u/time
+  /g/a/~zod/shell_terminal/u/child/4/main
+  /g/a/~zod/terminal/u/txt
+  /d/term-mess
+  //term/1
+]
+```
+
+This is the `duct` that the timer vane, Behn, receives when the `time` sample app asks the Behn to set a timer. This is also the `duct` over which the response is produced at the specified time. Unix sent a terminal keystroke event (enter), and Arvo routed it to Dill (our terminal), which passed it on to the Gall app terminal, which sent it to shell, its child, which created a new child (with process id 4), which on startup asked Behn to set a timer.
+
+Behn saves this `duct`, so that when the specified time arrives and Unix sends a
+wakeup event to the timer vane, it can produce the response on the same `duct`.
+This response is routed to the place we popped off the top of the `duct`, i.e. the
+time app. This app returns a `@d` which denotes the current time, which falls down to the shell,
+which drops it through to the terminal. Terminal drops this down to Dill, which
+converts it into an effect that Unix will recognize as a request to print the
+current time to the screen. When Dill produces this, the last path in the `duct` has an
+initial element of the empty span, so this is routed to Unix, which applies the effects.
+
+This is a call stack, with a crucial feature: the stack is a first-class citizen. You can respond over a `duct` zero, one, or many times. You can save `duct`s for later use. There are definitely parallels to Scheme-style continuations, but simpler and with more structure.
+
+
+#### `wire`
+
+```hoon
+++  wire  path                                          ::  event pretext
+```
+
+Synonym for `path`, used in `duct`s. These should be thought of as a list of symbols
+representing a cause.
+
+#### `move`
+
+```hoon
+++  move  [p=duct q=arvo]                               ::  arvo move
+```
+
+If `duct`s are a call stack, then how do we make calls and produce results? Arvo
+processes `move`s which are a combination of message data and metadata. There
+are three types of `move`s: `%pass`, `%give`, and `%unix`.
+
+A `%pass` `move` is analogous to a call:
+
+```
+[duct %pass return-path=path vane-name=@tD data=card]
+```
+
+Arvo pushes the return path (preceded by the first letter of the vane name) onto
+the `duct` and sends the given data, a `card`, to the vane we specified. Any response will come along the same `duct` with the `wire` `return-path`.
+
+A `%give` `move` is analogous to a return:
+
+```
+[duct %give data=card]
+```
+
+Arvo pops the top `wire` off the `duct` and sends the given `card` back to the caller.
+
+Lastly, a `%unix` `move` is how Arvo represents communication from Unix, such as a
+network request or terminal input.
+
+#### `card`s and `curd`s
+
+`card`s are the vane-specific portion of a `move`, while `curd`s are typeless
+`card`s utilized at the level of the kernel. `card`s are not actually defined in
```

Comment:
They're defined in zuse in +note-arvo and +sign-arvo (which refer to +task and +gift in each of the vane cores)